js_modules/dagit/packages/core/src/schedules/ScheduleBulkActionMenu.tsx

```diff
@@ -0,0 +1,81 @@
+import {Button, Icon, Menu, MenuItem, Popover} from '@dagster-io/ui';
+import * as React from 'react';
+
+import {InstigationStatus} from '../graphql/types';
+
+import {OpenWithIntent, ScheduleInfo, ScheduleStateChangeDialog} from './ScheduleStateChangeDialog';
+
+interface Props {
+  schedules: ScheduleInfo[];
+  onDone: () => void;
+}
+
+export const ScheduleBulkActionMenu = (props: Props) => {
+  const {schedules, onDone} = props;
+  const count = schedules.length;
+
+  const [openWithIntent, setOpenWithIntent] = React.useState<OpenWithIntent>('not-open');
+
+  const {anyOff, anyOn} = React.useMemo(() => {
+    let anyOff = false;
+    let anyOn = false;
+
+    for (const schedule of schedules) {
+      const {
+        scheduleState: {status},
+      } = schedule;
+      if (status === InstigationStatus.RUNNING) {
+        anyOn = true;
+      } else if (status === InstigationStatus.STOPPED) {
+        anyOff = true;
+      }
+      if (anyOn && anyOff) {
+        break;
+      }
+    }
+
+    return {anyOff, anyOn};
+  }, [schedules]);
```

Comment:
Not important at all, but this might be a fun use of `schedules.some(s => s.status === InstigationStatus.RUNNING)` since it does the early exit / returns false if the array is empty. Might do the trick here!

Reply:
I'm factoring this out for Sensors in a followup, will poke at this a bit more.